src/StudioCore/MsbEditor/Universe.cs

```diff
@@ -1557,6 +1561,79 @@ public void SaveAllMaps()
         }
     }
 
+    public void UpdateWorldMsbList()
+    {
+        try
+        {
+            if (_assetLocator.Type == GameType.EldenRing)
+            {
+                AssetDescription ad = _assetLocator.GetWorldLoadListList();
+                AssetDescription adw = _assetLocator.GetWorldLoadListList(true);
+                DCX.Type compressionType = GetCompressionType();
+
+                if (!Directory.Exists(Path.GetDirectoryName(adw.AssetPath)))
+                {
+                    Directory.CreateDirectory(Path.GetDirectoryName(adw.AssetPath));
+                }
+
+                // Write as a temporary file to make sure there are no errors before overwriting current file 
+                var worldMsbListPath = adw.AssetPath;
+                var worldMsbListPathTemp = adw.AssetPath + ".temp";
+
+                if (!File.Exists(worldMsbListPath))
+                {
+                    File.Copy(ad.AssetPath, worldMsbListPath, true);
+                }
+
+                // If a backup file doesn't exist of the original file create it
+                if (!File.Exists(worldMsbListPath + ".bak") && File.Exists(worldMsbListPath))
+                {
+                    File.Copy(worldMsbListPath, worldMsbListPath + ".bak", true);
+                }
+
+                if (File.Exists(worldMsbListPathTemp))
+                {
+                    File.Delete(worldMsbListPathTemp);
+                }
+
+                File.Copy(worldMsbListPath, worldMsbListPathTemp, true);
+
+                WORLDLOADLISTLIST p = WORLDLOADLISTLIST.Read(worldMsbListPathTemp);
+                IEnumerable<string> loadListMapIds = p.MapEntries.Where(me => me != null).Select(me => me.Id);
+                //For whatever reason, overworld maps (m60_xx_xx_xx) and skybox maps (mxx_xx_xx_99) are not included in the load list
+                IEnumerable<string> existingMapIds = LoadedObjectContainers.Keys.Where(em=>!(em.StartsWith("m60") || em.EndsWith("99")));
+                foreach (string id in existingMapIds)
+                {
+                    if (!loadListMapIds.Contains(id) && !(id.StartsWith("m60") || id.EndsWith("99")))
+                    {
+                        p.InsertNewMapEntry(id, false);
+                    }
+                }
+                p.Write(worldMsbListPathTemp, compressionType);
+
+                // Make a copy of the previous map
```

Comment:
Pretty sure we have a util for safely writing files we can use here

Reply:
Updated to use the safe util method.